libbpf-cargo/src/build.rs

```diff
@@ -270,6 +270,11 @@ pub fn build_single(
     } else {
         options.to_string()
     };
+
+    // Explicitly disable stack protector logic, which doesn't work with
+    // BPF. See https://lkml.org/lkml/2020/2/21/1000.
+    let compiler_options = format!("{compiler_options} -fno-stack-protector");
```

Comment:
Nit: avoid dropping the original options by mutating the original options. Also makes it more readable, but that's just my opinion

```suggestion
    compiler_options += " -fno-stack-protector";
```
or
```suggestion
    compiler_options.push_str(" -fno-stack-protector");
```

Reply:
Sure thing, done!